lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp

```diff
@@ -1802,24 +1794,20 @@ SwiftASTContext::CreateInstance(lldb::LanguageType language, Module &module,
     }
   }
 
-  if (!set_triple) {
-    llvm::Triple llvm_triple = swift_ast_sp->GetTriple();
-
-    // LLVM wants this to be set to iOS or MacOSX; if we're working on
-    // a bare-boards type image, change the triple for LLVM's benefit.
-    if (llvm_triple.getVendor() == llvm::Triple::Apple &&
-        llvm_triple.getOS() == llvm::Triple::UnknownOS) {
-      if (llvm_triple.getArch() == llvm::Triple::arm ||
-          llvm_triple.getArch() == llvm::Triple::thumb) {
-        llvm_triple.setOS(llvm::Triple::IOS);
-      } else {
-        llvm_triple.setOS(llvm::Triple::MacOSX);
-      }
-      swift_ast_sp->SetTriple(llvm_triple, &module);
+  // LLVM wants this to be set to iOS or MacOSX; if we're working on
+  // a bare-boards type image, change the triple for LLVM's benefit.
+  assert(triple == swift_ast_sp->GetTriple() && "triple was modifie");
+  if (triple.getVendor() == llvm::Triple::Apple &&
+      triple.getOS() == llvm::Triple::UnknownOS) {
+    if (triple.getArch() == llvm::Triple::arm ||
+        triple.getArch() == llvm::Triple::thumb) {
+      triple.setOS(llvm::Triple::IOS);
+    } else {
+      triple.setOS(llvm::Triple::MacOSX);
     }
+    swift_ast_sp->SetTriple(triple, &module);
```

Comment:
This logic looks fishy and dates back to 2015 (from bd28bd5a3b139d1f4cd3301bde71ef62185f61b8 the commit that imports swift support into lldb).

Is it valid anymore?

Reply:
It should be safe to remove it. The chances that an "unknown" platform would work even with this heuristic are close to zero.